### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW44)

### PR DESCRIPTION
**Pull request by preethi-intel**
LLVM: https://github.com/llvm/llvm-project/commit/cb088e8
SPIRV-LLVM-Translator: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/7a559fa

### CODE REVIEW

**Comment by haonanya** [comment]
Just look into several lit. 64 SPIRV lit failed on ubuntu2004_opaque_pointers_build_test since the lit are using old pointer type.

**Comment by pvchupin** [comment]
> Just look into several lit. 64 SPIRV lit failed on ubuntu2004_opaque_pointers_build_test since the lit are using old pointer type.

I think opaque pointers task issues could be ignored at this point. Adding @bader to comment.

**Comment by asudarsa** [comment]
Adding @jcranmer-intel for his comments. Thanks

**Comment by bader** [comment]
@preethi-intel, please, make sure that PR does not introduce new failures to existing jobs. You can find the results to compare with in Actions tab. Here is the link to the latest nightly - https://github.com/intel/llvm/actions/runs/3334289832/jobs/5517090344. 
I see that this PR introduces regressions to post-commit and nightly builds.

**Comment by jcranmer-intel** [comment]
> @preethi-intel, please, make sure that PR does not introduce new failures to existing jobs. You can find the results to compare with in Actions tab. Here is the link to the latest nightly - https://github.com/intel/llvm/actions/runs/3334289832/jobs/5517090344. I see that this PR introduces regressions to post-commit and nightly builds.

From what I see, the new llvm-spirv lit test failures in opaque mode are transcoding/multiple_user_semantic{,_nopaque}.ll, which were newly added in the pulldown, so it's not causing any regressions to llvm-spirv tests.

**Comment by bader** [comment]
> I see that this PR introduces regressions to post-commit and nightly builds.

I'm referring to `check-clang` failures here.

**Comment by preethi-intel** [comment]
> > I see that this PR introduces regressions to post-commit and nightly builds.
> 
> I'm referring to `check-clang` failures here.

@bader We have opened JIRA for the team to look into these check-clang fails.

File: clang/lib/Sema/SemaType.cpp [hunk 1]
```diff
@@ -2727,8 +2731,15 @@ QualType Sema::BuildVectorType(QualType CurType, Expr *SizeExpr,
...
 
   if (!TypeSize || VectorSizeBits % TypeSize) {
-    Diag(AttrLoc, diag::err_attribute_invalid_size)
-        << SizeExpr->getSourceRange();
+    // Disable errors for SYCL and OpenMP device since definition of __bf16 is
+    // being moved to a shared header and it causes new errors emitted when
+    // host code is compiled with device compiler for SPIR target.
+    // FIXME: device code specific diagnostic is probably needed.
+    if (!(!TypeSize &&
+          (getLangOpts().OpenMPIsDevice || getLangOpts().SYCLIsDevice))) {
```
Comment (Fznamznon) [inline clang/lib/Sema/SemaType.cpp:2739]: Can you double check if `OpenMPIsDevice` switch is required here? It may not.

Comment (pvchupin) [reply clang/lib/Sema/SemaType.cpp:2739]: @preethi-intel, have you open a ticket for this FIXME to follow up?

**Comment by pvchupin** [comment]
\merge

File: clang/lib/Sema/SemaType.cpp (hunk 1 above)
Comment (Fznamznon) [reply clang/lib/Sema/SemaType.cpp:2739]: @pvchupin , I have opened an internal tracker for this FIXME.

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Mon 31 Oct 2022 07:34:54 PM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bb-sycl** [comment]
Mon 31 Oct 2022 07:44:18 PM UTC --- Merge the branch in this PR to base automatically. Will close the PR later.

**Pull request merged**